### PR TITLE
llvm@6: Fix build for Linuxbrew

### PR DESCRIPTION
**Pull request by iMichka**

### CODE REVIEW

**Comment by iMichka** [comment]
```
llvm@6:
  * C: 55: col 5: FormulaAudit/Conflicts: Versioned formulae should not use `conflicts_with`. Use `keg_only :versioned_formula` instead.
==> FAILED

==> brew style llvm@6
== /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/llvm@6.rb ==
C:209:  4: Layout/IndentationConsistency: Inconsistent indentation detected.
C:210:  4: Layout/IndentationWidth: Use 2 (not 3) spaces for indentation.
W:212:  5: Layout/EndAlignment: end at 212, 4 is not aligned with unless OS.mac? at 209, 3.
W:224: 51: Lint/RequireParentheses: Use parentheses in the method call to avoid confusion about precedence.
C:236:  1: Layout/TrailingWhitespace: Trailing whitespace detected.
```

**Comment by iMichka** [comment]
```
==> brew test llvm@6 --verbose
Testing llvm@6
==> /home/linuxbrew/.linuxbrew/Cellar/llvm@6/6.0.1/bin/llvm-config --prefix
==> /home/linuxbrew/.linuxbrew/Cellar/llvm@6/6.0.1/bin/clang -L/home/linuxbrew/.linuxbrew/Cellar/llvm@6/6.0.1/lib -fopenmp -nobuiltininc -I/home/linuxbrew/.linuxbrew/Cellar/llvm@6/6.0.1/lib/clang/6.0.1/include omptest.c -o omptest

==> ./omptest
./omptest: error while loading shared libraries: libomp.so: cannot open shared object file: No such file or directory
Error: llvm@6: failed
An exception occured within a build process:
  Test::Unit::AssertionFailedError: <0> expected but was
<127>.
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.7/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/assertions.rb:55:in `block in assert_block'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.7/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/assertions.rb:1632:in `_wrap_assertion'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.7/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/assertions.rb:53:in `assert_block'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.7/lib/ruby/gems/2.3.0/gems/test-unit-3.1.5/lib/test/unit/assertions.rb:240:in `assert_equal'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula_assertions.rb:10:in `shell_output'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/llvm@6.rb:279:in `block in <class:LlvmAT6>'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1696:in `block (3 levels) in run_test'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils.rb:520:in `with_env'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1695:in `block (2 levels) in run_test'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:861:in `with_logging'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1694:in `block in run_test'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1933:in `block in mktemp'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/mktemp.rb:55:in `block in run'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/mktemp.rb:55:in `chdir'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/mktemp.rb:55:in `run'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1932:in `mktemp'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1688:in `run_test'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/test.rb:28:in `block in <main>'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.7/lib/ruby/2.3.0/timeout.rb:91:in `block in timeout'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.7/lib/ruby/2.3.0/timeout.rb:33:in `block in catch'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.7/lib/ruby/2.3.0/timeout.rb:33:in `catch'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.7/lib/ruby/2.3.0/timeout.rb:33:in `catch'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.7/lib/ruby/2.3.0/timeout.rb:106:in `timeout'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/test.rb:27:in `<main>'
==> FAILED
```

File: Formula/llvm@6.rb [hunk 1]
```diff
@@ -198,6 +245,18 @@ def install
     # install llvm python bindings
     (lib/"python2.7/site-packages").install buildpath/"bindings/python/llvm"
     (lib/"python2.7/site-packages").install buildpath/"tools/clang/bindings/python/clang"
+
+    unless OS.mac?
+      # Remove conflicting libraries.
+      # libgomp.so conflicts with gcc.
+      rm lib/"libgomp.so"
```
Comment (xu-cheng) [inline Formula/llvm@6.rb]: This line may be removed since this is a versioned (hence keg-only) formula.

Comment (xu-cheng) [inline Formula/llvm@6.rb]: This could be removed as this is a keg-only formula

**Comment by iMichka** [comment]
```
CMake Warning:
  Manually-specified variables were not used by the project:

    LLVM_ENABLE_LIBCXXABI
```

```
-- Performing Test CXX_SUPPORTS_STDLIB
-- Performing Test CXX_SUPPORTS_STDLIB - Failed
CMake Warning at cmake/modules/HandleLLVMStdlib.cmake:24 (message):
  Can't specify libc++ with '-stdlib='
Call Stack (most recent call first):
  cmake/config-ix.cmake:16 (include)
  CMakeLists.txt:578 (include)
```

**Pull request closed without merging**